lldb/include/lldb/API/SBTarget.h

```diff
@@ -986,6 +987,42 @@ class LLDB_API SBTarget {
 
   lldb::SBMutex GetAPIMutex() const;
 
+  /// Register a scripted frame provider for this target.
+  /// If a scripted frame provider with the same name is already registered on
+  /// this target, it will be overwritten.
+  ///
+  /// \param[in] class_name
+  ///     The name of the Python class that implements the frame provider.
+  ///
+  /// \param[in] args_dict
+  ///     A dictionary of arguments to pass to the frame provider class.
+  ///
+  /// \param[in] thread_filter
```

Comment:
How do you handle the case where when you register the provider some or all of the threads you want it to match against haven't been created yet?  I was expecting this to be algorithmic.

Reply:
How can you know the thread name of id or a thread that haven't yet been created ? Also if we wanted to go that route, we would have to introduce a new SBAPI class.

Reply:
That's an option but it's not very optimal since we'll be creating python instances of the frame provider just to ask if we need to associate it with said thread and discard it otherwise. Seems a bit overkill.

Reply:
You don't need to create an instance of a python class to call a static method, do you?

Reply:
Ok, after some offline chatting, we came to the conclusion that the static method would be a good way to let the user take care of the filtering themselves but it might still be useful to be able to do it from the command line. We should just remove it in the SBAPI.